core/src/main/java/org/jdbi/v3/core/collector/MapCollectorFactory.java

```
@@ -0,0 +1,99 @@
+            .put(Map.class, toMap(LinkedHashMap::new))
+            .put(LinkedHashMap.class, toMap(LinkedHashMap::new))
+            .put(HashMap.class, toMap(HashMap::new))
+            .put(TreeMap.class, toMap(TreeMap::new))
+            .put(SortedMap.class, toMap(TreeMap::new))
+            .put(ConcurrentMap.class, toMap(ConcurrentHashMap::new))
+            .put(ConcurrentHashMap.class, toMap(ConcurrentHashMap::new))
+            .put(WeakHashMap.class, toMap(WeakHashMap::new))
+            .build();
+
+    @Override
+    public boolean accepts(Type containerType) {
+        return collectors.containsKey(getErasedType(containerType))
+                && (containerType instanceof ParameterizedType); // no raw Map
+    }
+
+    @Override
+    public Optional<Type> elementType(Type containerType) {
+        TypeToken<?> keyType = TypeToken.of(containerType).resolveType(KEY_PARAM);
+        TypeToken<?> valueType = TypeToken.of(containerType).resolveType(VALUE_PARAM);
+        return Optional.of(entryType(keyType, valueType));
+    }
+
+    private <K, V> Type entryType(TypeToken<K> keyType, TypeToken<V> valueType) {
+        return new TypeToken<Map.Entry<K, V>>() {}
+                .where(new TypeParameter<K>() {}, keyType)
+                .where(new TypeParameter<V>() {}, valueType)
+                .getType();
+    }
+
+    @Override
+    public Collector<?, ?, ?> build(Type containerType) {
+        return collectors.get(getErasedType(containerType));
+    }
+
+    static <K, V, M extends Map<K, V>> Collector<Map.Entry<K,V>, ?, M> toMap(Supplier<M> supplier) {
+        return Collector.of(
+                supplier,
+                MapCollectorFactory::putEntry,
+                MapCollectorFactory::combine);
+    }
+
+    private static <K, V, M extends Map<K, V>> void putEntry(M map, Map.Entry<K, V> entry) {
+        // TODO throw exception if duplicate entry?
```

I think it would be better to throw an exception on duplicates rather than silently swallowing them. In that case the user will be encouraged to switch from `Map` to `Multimap`, if several values for one key are possible.